src/test/java/org/broadinstitute/hellbender/tools/reference/CompareReferencesIntegrationTest.java

```diff
@@ -118,19 +119,109 @@ public void testCompareReferencesMultipleReferencesStdOut() throws IOException{
         final File ref3 = new File(getToolTestDataDir() + "hg19mini_chr2snp.fasta");
         final File ref4 = new File(getToolTestDataDir() + "hg19mini_missingchr1.fasta");
 
-        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref2.getAbsolutePath(), "-refcomp", ref3.getAbsolutePath(),
-                "-refcomp", ref4.getAbsolutePath()};
+        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref3.getAbsolutePath(), "-refcomp", ref3.getAbsolutePath(),
+                "-refcomp", ref4.getAbsolutePath(), "-display-sequences-by-name", "-display-only-differing-sequences"};
         runCommandLine(args);
     }
 
-    @Test(enabled = false)
-    public void testCompareReferencesMissingSequencesStdOut() throws IOException{
+    // FIND_SNPS_ONLY tests:
+    @Test
+    public void testFindSNPsMultipleSNPs() throws IOException{
+        final File ref1 = new File(getToolTestDataDir() + "hg19mini.fasta");
+        final File ref2 = new File(getToolTestDataDir() + "hg19mini_chr2multiplesnps.fasta");
+        final File output = IOUtils.createTempDir("tempFindSNPs");
+
+        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref2.getAbsolutePath(), "-base-comparison", "FIND_SNPS_ONLY", "-base-comparison-output", output.getAbsolutePath()};
+        runCommandLine(args);
+
+        final File actualOutput = new File(output, "hg19mini.fasta_hg19mini_chr2multiplesnps.fasta_snps.tsv");
+        final File expectedOutput = new File(getToolTestDataDir(), "expected.hg19mini.fasta_hg19mini_chr2multiplesnps.fasta_snps.tsv");
+        IntegrationTestSpec.assertEqualTextFiles(actualOutput, expectedOutput);
+    }
+
+    @Test
+    public void testFindSNPsIUPACBases() throws IOException{
         final File ref1 = new File(getToolTestDataDir() + "hg19mini.fasta");
-        final File ref2 = new File(getToolTestDataDir() + "hg19mini_missingchr3.fasta");
-        final File ref3 = new File(getToolTestDataDir() + "hg19mini_missingchr1.fasta");
+        final File ref2 = new File(getToolTestDataDir() + "hg19mini_chr2iupacsnps.fasta");
+        final File output = IOUtils.createTempDir("tempFindSNPs");
 
-        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref2.getAbsolutePath(), "-refcomp", ref3.getAbsolutePath()};
+        final String[] args = new String[] {"-R", ref1.getAbsolutePath() , "-refcomp", ref2.getAbsolutePath(), "-base-comparison", "FIND_SNPS_ONLY", "-base-comparison-output", output.toPath().toString()};
         runCommandLine(args);
+
+        final File expectedOutput = new File(getToolTestDataDir(), "expected.hg19mini.fasta_hg19mini_chr2iupacsnps.fasta_snps.tsv");
+        final File actualOutput = new File(output, "hg19mini.fasta_hg19mini_chr2iupacsnps.fasta_snps.tsv");
+
+        IntegrationTestSpec.assertEqualTextFiles(actualOutput, expectedOutput);
     }
 
+    // FULL_ALIGNMENT tests:
```

Comment:
These FULL_ALIGNMENT mode tests could be unified using a `DataProvider`, since the actual test code is the same in every case. The `DataProvider` would list the two fasta inputs, plus the expected output for each test case.